src/reader.rs

```diff
@@ -618,6 +618,7 @@
         if !eval_res.no_status {
             STATUS_COUNT.fetch_add(1, Ordering::Relaxed);
         }
+        data.add_to_history(command.clone(), parser);
```

Comment:
We need to think about whether the behavior change is worth it.
Logging the command before running it seems like a safe default behavior,
which makes a difference in some cases like `exec` you mention.

If I start a long-running command, open a new terminal and press Up-arrow
before the first command is finished, what should the behavior be?
Maybe it doesn't matter maybe it does.

I think we don't have to decide now; we can err on the side of caution and merge this without support for querying
`$status` etc, especially since no one explicitly asked for that.
I think `$status` is not super relevant because most failed commands will be shadowed by their fixed versions.
Users who use a history pager might see things differently.. but only if they configure this behavior, so I wouldn't care too much.
Intuitively I'd tell users who want to use `$status` to remove those command from history in a `postexec` hook?

Reply:
> We need to think about whether the behavior change is worth it.

Yup.

> $status etc, especially since no one explicitly asked for that.

We have had people ask for it, e.g. #5 is relevant.

[sponge](https://github.com/meaningful-ooo/sponge) features it.

If $status isn't available, there's the question what fish_should_add_history would in practice do. If it's just matching against a list or maybe regex we could also get away with a variable.

Would people check against, say, $PWD?